cosmwasm/ibc-union/core/msg/src/lightclient.rs

```diff
@@ -32,12 +24,29 @@ pub enum VerifyCreationResponseEvent {
     },
 }
 
+pub type StorageWrites = Vec<(Bytes, Bytes)>;
+
 #[derive(serde::Serialize, serde::Deserialize, Debug)]
 #[serde(deny_unknown_fields, rename_all = "snake_case")]
 pub struct VerifyCreationResponse {
     pub latest_height: u64,
     pub counterparty_chain_id: String,
-    pub events: Option<Vec<VerifyCreationResponseEvent>>,
+    pub client_state_bytes: Option<Bytes>,
+    pub storage_writes: StorageWrites,
+    pub events: Vec<VerifyCreationResponseEvent>,
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+#[serde(deny_unknown_fields, rename_all = "snake_case")]
+pub struct UpdateStateResponse {
+    /// The height to save the consensus state at
+    pub height: u64,
+    /// The client state to overwrite the current one with if provided
+    pub client_state_bytes: Option<Bytes>,
```

Comment:
same comment here about always storing it

Reply:
This actually makes you able to save two writes when it's not needed (commitment and client state store). I kinda prefer every bit of gas optimizations whenever we can. But if you are strongly against it, we can change.

Reply:
makes sense. as long as it's well documented it should be fine